### PR TITLE
 Clojure example for fixed label-width captcha recognition 

### PR DESCRIPTION
**Pull request by kedarbellare**
## Description ##

Adds an example for training a captcha OCR model and using the learned model for performing inference.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [X] Changes are complete (i.e. I finished coding on this PR)
- [X] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [X] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [X] Captcha OCR training and inference with unit tests for training

## Comments ##
I ran this on both osx and ubuntu linux but only with `:cpu`. However, I encountered memory issues while trying to train on a `:gpu`:
```
Exception in thread "main" org.apache.mxnet.MXNetError: [19:18:33] src/storage/./pooled_storage_manager.h:143: cudaMalloc failed: out of memory
```

## Reviewers ##

@gigasquid

### CODE REVIEW

**Comment by gigasquid** [comment]
Cool! Excited to look at this closer 😸

**Comment by Roshrini** [comment]
@kedarbellare Thanks for this contribution! On which gpu machine, did you run this example?

@mxnet-label-bot Add [pr-awaiting-review, Clojure]

**Comment by kedarbellare** [comment]
@Roshrini : I ran this on my personal desktop with Nvidia GTX 1070 with 8GB memory (ubuntu 18.04, cuda-9.0). I tried varying the batch sizes but that didn't help. Let me know if you want more details.

**Comment by gigasquid** [comment]
@kedarbellare - getting to try this out - how long did your training take locally (how many epochs) ?

File: contrib/clojure-package/examples/captcha/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+WARN  org.apache.mxnet.DataDesc: Found Undefined Layout, will use default index 0 for batch axis
+INFO  org.apache.mxnet.infer.Predictor: Latency increased due to batchSize mismatch 8 vs 1
+WARN  org.apache.mxnet.DataDesc: Found Undefined Layout, will use default index 0 for batch axis
+WARN  org.apache.mxnet.DataDesc: Found Undefined Layout, will use default index 0 for batch axis
+CAPTCHA output: 6643
+INFO  org.apache.mxnet.util.NativeLibraryLoader: Deleting /tmp/mxnet6045308279291774865/libmxnet.so
+INFO  org.apache.mxnet.util.NativeLibraryLoader: Deleting /tmp/mxnet6045308279291774865/mxnet-scala
+INFO  org.apache.mxnet.util.NativeLibraryLoader: Deleting /tmp/mxnet6045308279291774865
+```
+The model is run on `captcha_example.png` by default.
```
Comment (aaronmarkham) [inline contrib/clojure-package/examples/captcha/README.md]:
```suggestion
The model runs on `captcha_example.png` by default.
```

**Comment by kedarbellare** [comment]
It's pretty slow with CPU training. On my macbook pro, with `lein train` (default one CPU training) the speedometer shows ~25 samples/sec (end-to-end training took over an hour for 10 epochs). My beefier deskop gets double that with CPU training and GPU training shows over 600 samples/sec.

**Comment by kedarbellare** [comment]
@aaronmarkham I followed your suggestions for the README. I'll add more comments in the code in a subsequent commit.

File: contrib/clojure-package/examples/captcha/src/captcha/train_ocr.clj [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+                               ndarray/->vec)]
+    (if by-character
+      (float (/ digit-matches nr nc))
+      (float (/ complete-matches nr)))))
+
+(defn get-data-symbol
+  []
+  (let [data (sym/variable "data")
+        ;; normalize the input pixels
+        scaled (sym/div (sym/- data 127) 128)
```
Comment (gigasquid) [inline contrib/clojure-package/examples/captcha/src/captcha/train_ocr.clj]: Great way to normalize the pixels 💯

**Pull request merged**